modulemd/v2/modulemd-module-stream-v2.c

```diff
@@ -106,6 +106,130 @@ modulemd_module_stream_v2_finalize (GObject *object)
+
+  if (!MODULEMD_MODULE_STREAM_CLASS (modulemd_module_stream_v2_parent_class)
+         ->equals (self_1, self_2))
+    {
+      return FALSE;
+    }
+
+  /*Check property equality*/
+  if (g_strcmp0 (v2_self_1->community, v2_self_2->community) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v2_self_1->description, v2_self_2->description) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v2_self_1->documentation, v2_self_2->documentation) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v2_self_1->summary, v2_self_2->summary) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v2_self_1->tracker, v2_self_2->tracker) != 0)
+    return FALSE;
+
+  if (!modulemd_buildopts_equals (v2_self_1->buildopts, v2_self_2->buildopts))
+    return FALSE;
+
+  if (!modulemd_hash_table_equals (v2_self_1->rpm_components,
+                                   v2_self_2->rpm_components,
+                                   modulemd_component_equals_wrapper))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (v2_self_1->module_components,
+                                   v2_self_2->module_components,
+                                   modulemd_component_equals_wrapper))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v2_self_1->module_licenses,
+                                           v2_self_2->module_licenses))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v2_self_1->content_licenses,
+                                           v2_self_2->content_licenses))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (v2_self_1->profiles,
+                                   v2_self_2->profiles,
+                                   modulemd_profile_equals_wrapper))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v2_self_1->rpm_api,
+                                           v2_self_2->rpm_api))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v2_self_1->rpm_artifacts,
+                                           v2_self_2->rpm_artifacts))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v2_self_1->rpm_filters,
+                                           v2_self_2->rpm_filters))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (v2_self_1->servicelevels,
+                                   v2_self_2->servicelevels,
+                                   modulemd_service_level_equals_wrapper))
+    {
+      return FALSE;
+    }
+
+
+  /* < string, GHashTable <string, Modulemd.RpmMapEntry> > */
+  if (!modulemd_hash_table_equals (
+        v2_self_1->rpm_artifact_map,
+        v2_self_2->rpm_artifact_map,
+        modulemd_RpmMapEntry_hash_table_equals_wrapper))
+    {
+      return FALSE;
+    }
+
+
+  if (v2_self_1->dependencies->len != v2_self_2->dependencies->len)
+    return FALSE;
+
+  for (guint i = 0; i < v2_self_1->dependencies->len; i++)
```

Comment:
Please include a comment here explaining that ordering is important for the dependencies, so each array index must be the same.